azurelinuxagent/agent.py

```diff
@@ -159,7 +176,7 @@ def main(args=[]): # pylint: disable=R0912,W0102
     """
     if len(args) <= 0: # pylint: disable=len-as-condition
         args = sys.argv[1:]
-    command, force, verbose, debug, conf_file_path = parse_args(args)
+    command, force, verbose, debug, conf_file_path, log_collector_mode = parse_args(args)
```

Comment:
This seems like we're returning a lot of arguments, and might need to return more later on (with the addition of more options). Would the return type be better as a specific class, or maybe a dict?

Reply:
It would be cleaner, but I will leave that change out from this PR since it's tangential to the feature. I also don't really expect this to grow in the future, to be honest.

Reply:
+1 on @kevinclark19a's point. This would definitely make it cleaner and would also make our tests more cleaner (all the tests in `test_agent.py` where we are returning all args even if we dont need it - 
Eg: 
`c, f, v, d, cfp, lcm = parse_args(["-collect-logs", "-mode:invalid"])`